docs-src/references/dynamic-configuration.md

```diff
@@ -0,0 +1,209 @@
+
+The dynamic configuration keys are set with default values when you create your Cluster configuration.
+You can override these values as you test your Cluster setup for optimal performance according to your workload requirements.
+
+For the complete list of dynamic configuration keys, see <https://github.com/temporalio/temporal/blob/master/common/dynamicconfig/constants.go>.
+Ensure that you check server releases notes for any changes to these keys and values.
+
+To check the default values set for a dynamic configuration key, check the following links:
+
+- [Frontend Service](https://github.com/temporalio/temporal/blob/5783e781504d8ffac59f9848b830868f3139b980/service/frontend/service.go#L176)
+- [History Service](https://github.com/temporalio/temporal/blob/5783e781504d8ffac59f9848b830868f3139b980/service/history/configs/config.go#L309)
+- [Matching Service](https://github.com/temporalio/temporal/blob/5783e781504d8ffac59f9848b830868f3139b980/service/matching/config.go#L125)
+- [Worker Service](https://github.com/temporalio/temporal/blob/5783e781504d8ffac59f9848b830868f3139b980/service/worker/service.go#L193)
+
+Note that setting dynamic configuration is optional.
+Change these values only if you need to override the default values to achieve better performance on your Temporal Cluster.
+Also, ensure that you test your changes before setting these in production.
+
+## Format
+
+To override the default dynamic configuration values, specify your custom values and constraints for the dynamic configuration keys that you want to change in a YAML configuration file.
+Use the following format when creating your dynamic configuration file.
+
+```yaml
+testGetBoolPropertyKey:
+  - value: false
+  - value: true
+    constraints:
+      namespace: "your-namespace"
+  - value: false
+    constraints:
+      namespace: "your-other-namespace"
+testGetDurationPropertyKey:
+  - value: "1m"
+    constraints:
+      namespace: "your-namespace"
+      taskQueueName: "longIdleTimeTaskqueue"
+testGetFloat64PropertyKey:
+  - value: 12.0
+    constraints:
+      namespace: "your-namespace"
+testGetMapPropertyKey:
+  - value:
+      key1: 1
+      key2: "value 2"
+      key3:
+        - false
+        - key4: true
+          key5: 2.0
+```
+
+### Constraints
+
+You can define constraints on some dynamic configuration keys to set specific values that apply on a Namespace or Task Queue level.
+Not defining constraints on a dynamic configuration key sets the values globally across the Cluster.
+
+- To set global values for the configuration key with no constraints, use:
+
+  ```yaml
+  frontend.globalNamespaceRPS: # Namespace rate limit per second applied globally on the entire Cluster.
+    - value: 5000 # There is no default limit. If set, this value overrides frontend.namespacerps.
```

Comment:
```suggestion
  frontend.globalNamespaceRPS: # Total per-Namespace RPC rate limit across the Cluster
    - value: 5000
```
So, one confusing thing here is that "global" in the sentence above this and "global" in this key name mean two totally different things. The first is "not specific to any namespace or task queue" and the second is "total across all frontend instances".

So I would avoid using "global" in the comment describing "globalNamespaceRPS" because it's potentially confusing. You can just say "across the cluster".

I'd also just remove the comment about the default, I don't think it adds much here. At least remove the bit about "if set, overrides ..." because we're documenting the dynamic config system here, not specific keys.

Reply:
updated